packages/kit/src/core/sync/write_tsconfig.js

```diff
@@ -85,19 +85,28 @@ export function get_tsconfig(kit, include_base_url) {
 	/** @param {string} file */
 	const config_relative = (file) => posixify(path.relative(kit.outDir, file));
 
-	const include = ['ambient.d.ts', './types/**/$types.d.ts', config_relative('vite.config.ts')];
-	for (const dir of [kit.files.routes, kit.files.lib]) {
-		const relative = project_relative(path.dirname(dir));
-		include.push(config_relative(`${relative}/**/*.js`));
-		include.push(config_relative(`${relative}/**/*.ts`));
-		include.push(config_relative(`${relative}/**/*.svelte`));
+	const include = new Set([
+		'ambient.d.ts',
+		'./types/**/$types.d.ts',
+		config_relative('vite.config.ts')
+	]);
+
+	for (const dir of [
+		kit.files.routes,
+		kit.files.lib,
+		path.resolve('src') // TODO(v2): remove src and only scope it to what's necessary
+	]) {
```

Comment:
```suggestion
	const src_includes = [
		kit.files.routes,
		kit.files.lib,
		path.resolve('src') // TODO(v2): find a better way to include all src files
	].filter((dir) => !path.relative(dir, path.resolve('src')).startsWith('..'));
	for (const dir of src_includes) {
```

Reply:
Looks like the check is wrong and the tests are failing. I can never remember the right order of parent/child and whether to use the not for this.

But I think `!path.relative(dir, path.resolve('src')).endsWith('..')` would work,

dir = src/lib/something => "../..",
dir = src/routes => "..",
dir = src => "",
dir = app => "../src"

am I missing any case?